chart/crds/crds.yaml

```diff
@@ -54,11 +54,42 @@ spec:
           spec:
             description: ControlPlaneSpec defines the desired state of ControlPlane
             properties:
+              adoptedTokenExpirationSeconds:
+                description: expiration time for token of adopted cluster
+                format: int64
+                type: integer
               backend:
                 enum:
                 - shared
                 - dedicated
                 type: string
+              bootstrapSecretRef:
+                description: |-
+                  BootstrapSecretRef contains a reference to the kubeconfig used to bootstrap adoption of
+                  an external cluster
+                properties:
+                  inClusterKey:
```

Comment:
Can you please clarify what is `inClusterKey`?

Perhaps I missed, but I did not see a documentation of what the bootstrap secret should look like. This is useful if somebody wants/needs to create it without using kflex.

Reply:
You may see the defaults being used for the bootstrap secret in `cmd/kflex/adopt/adopt.go`. The name is defined in:
```go
func GenerateBoostrapSecretName(cpName string) string {
	return fmt.Sprintf("%s-bootstrap", cpName)
}
```
namespace is `SystemNamespace  = "kubeflex-system"` and key is `KubeconfigSecretKeyInCluster  = "kubeconfig-incluster"`

Reply:
To further clarify, `inClusterKey` is the name of the secret generated by kflex from the bootstrap secret... right?

Since you have a default, can you make it not required?

Reply:
> To further clarify, inClusterKey is the name of the secret generated by kflex from the bootstrap secret
No, `inClusterKey` is the name of the key used for the kubeconfig, which is `kubeconfig-incluster`. The name of the secret would be `<cp-name>-bootstrap`. You do not have to provide these values in the ControlPlane CR, they are optional.

Reply:
(A) Since the golang source re-uses the existing type `SecretReference`, regularity is implied here. It is a reference to the same sort of thing. Same schema for contents. I am not sure yet whether this usage really is regular.

(B) @pdettori, what do you mean by "You do not have to provide these values in the ControlPlane CR, they are optional"? Which values are optional?